src/__tests__/decode.test.ts

```diff
@@ -219,7 +227,7 @@ test.each(cases)(
   { timeout: 30_000 },
   (name, file, image) => {
     const result = decode(image);
-    expect(result).toHaveLength(1);
+    expect(result).toHaveLength(name === 'dog.tiff' ? 8 : 1);
```

Comment:
To avoid this:
- Update `TiffFile` to add a new optional field with the number of ifds (could be named `pages` for example).
- Add this field for `dog.tiff`
- Update `cases` to add this number in the mapped array: `(file) => [file.name, file, readImage(file.name), file.pages ?? 1] as const,`
- Use it here: `expect(result).toHaveLength(pages);`